var/spack/repos/builtin/packages/mfem/package.py

```diff
@@ -697,6 +699,29 @@ def install(self, spec, prefix):
         if install_em:
             install_tree('data', join_path(prefix_share, 'data'))
 
+    examples_src_dir = 'examples'
+    examples_data_dir = 'data'
+
+    @run_after('install')
+    def cache_test_sources(self):
+        """Copy the example source files after the package is installed to an
+        install test subdirectory for use during `spack test run`."""
+        self.cache_extra_test_sources(self.examples_src_dir)
+        self.cache_extra_test_sources(self.examples_data_dir)
+
+    def test(self):
+        test_dir = os.path.join(self.install_test_root, self.examples_src_dir)
+        with working_dir(test_dir, create=False):
+            test_exe = 'ex10p' if ('+mpi' in self.spec) else 'ex10'
```

Comment:
Is there any rationale for why example 10 was chosen? Is it a particularly stripped down example? Is it known to run no matter how MFEM is configured? Does it exercise something special in MFEM? If the choice is completely arbitrary or if there is a specific reason for the choice, lets make a comment here explaining that choice.

Reply:
We'd have to check with the E4S team for why they chose that example.

Sergei was only asked to add their validation test(s) to the package.

Reply:
I guess I neglected to submit my final comment. 

I think something to the effect...

```
MFEM has many examples to serve as a suitable smoke check. ex10 was chosen arbitrarily.
```

Is fine.

Reply:
@tldahlgren @markcmiller86 I was actually mirroring the E4S test suite and they used `ex10`. My best guess is that it's one of the few examples that has both serial `ex10` and parallel `ex10p` versions